test/OpenSpaceToolkit/Astrodynamics/Trajectory/Validation/Propagator.cross.validation.cpp

```diff
@@ -0,0 +1,1584 @@
+{
+    // EGM96 70x70 perturbation only vs STK WGS84EGM96
+    {
+        // Current state and instant setup
+        const Instant startInstant = Instant::DateTime(DateTime::Parse("2021-03-20 00:00:00.000"), Scale::UTC);
+
+        // Reference data setup
+        const Table referenceData = Table::Load(
+            File::Path(Path::Parse("/app/test/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/"
+                                   "Propagated/STK_WGS84EGM96_70x70_2hr_run.csv")),
+            Table::Format::CSV,
+            true
+        );
+
+        Array<Instant> instantArray = Array<Instant>::Empty();
+        Array<Vector3d> referencePositionArrayGCRF = Array<Vector3d>::Empty();
+        Array<Vector3d> referenceVelocityArrayGCRF = Array<Vector3d>::Empty();
+
+        for (const auto& referenceRow : referenceData)
+        {
+            instantArray.add(startInstant + Duration::Seconds(referenceRow[1].accessReal()));
+
+            referencePositionArrayGCRF.add(
+                1e3 * Vector3d(referenceRow[2].accessReal(), referenceRow[3].accessReal(), referenceRow[4].accessReal())
+            );
+            referenceVelocityArrayGCRF.add(
+                1e3 * Vector3d(referenceRow[5].accessReal(), referenceRow[6].accessReal(), referenceRow[7].accessReal())
+            );
+        }
+
+        // Create dynamics
+        const Shared<Celestial> earth = std::make_shared<Celestial>(Earth::EGM96(70, 70));
+        const Array<Shared<Dynamics>> dynamics = {
+            std::make_shared<PositionDerivative>(),
+            std::make_shared<CentralBodyGravity>(earth),
+        };
+
+        // Setup Propagator
+        const Propagator propagator = {defaultNumericalSolver_, dynamics};
+
+        // Setup initial conditions
+        const State state = {
+            startInstant,
+            Position::Meters({referencePositionArrayGCRF[0]}, gcrfSPtr_),
+            Velocity::MetersPerSecond({referenceVelocityArrayGCRF[0]}, gcrfSPtr_),
+        };
+
+        // Propagate all states
+        const Array<State> propagatedStateArray = propagator.calculateStatesAt(state, instantArray);
+
+        // Validation loop
+        for (size_t i = 0; i < instantArray.getSize(); i++)
+        {
+            // GCRF Compare
+            const Position positionGCRF = propagatedStateArray[i].getPosition();
+            const Velocity velocityGCRF = propagatedStateArray[i].getVelocity();
+
+            const double positionErrorGCRF = (positionGCRF.accessCoordinates() - referencePositionArrayGCRF[i]).norm();
+            const double velocityErrorGCRF = (velocityGCRF.accessCoordinates() - referenceVelocityArrayGCRF[i]).norm();
+
+            ASSERT_EQ(*Frame::GCRF(), *positionGCRF.accessFrame());
+            ASSERT_EQ(*Frame::GCRF(), *velocityGCRF.accessFrame());
+
+            ASSERT_GT(1.5e-4, positionErrorGCRF);
+            ASSERT_GT(1.5e-7, velocityErrorGCRF);
+
+            // Results console output
```

Comment:
Leftover?

(not gonna flag any more of these, if they are a leftover, maybe you can search and replace in bulk)